### PR TITLE
[Service/PkgMgr] Add skeleton code for checking installed resource packages @open sesame 2/21 19:00

### PR DESCRIPTION
**Pull request by again4you**
This patch checks all installed resource packages and registers them into the database if they are not registered.

Signed-off-by: Sangjung Woo <sangjung.woo@samsung.com>

### Basic Operation
```bash
root:~> /usr/bin/machine-learning-agent 
init gstreamer
Installed count: 1
Pkg name: org.tizen.mlmodel
- file: mobilenet_v2_1.0_224.tflite
- file: model_description.xml
```

### CODE REVIEW

File: daemon/pkg-mgr.c [hunk 1]
```diff
@@ -85,6 +85,96 @@ static void _pkg_mgr_event_cb (const char *type, const char *package,
...
+  dir = g_dir_open (pkg_path, 0, NULL);
+  if (dir) {
+    const gchar *file_name;
+    while ((file_name = g_dir_read_name (dir))) {
+      _I ("- file: %s", file_name);
+    }
+    g_dir_close (dir);
+  }
+
+  ret = package_info_destroy (handle);
```
Comment (jaeyun-jung) [inline daemon/pkg-mgr.c:126]: Q: should we destroy `package_info_h handle` here, not outside of the callback?

File: daemon/pkg-mgr.c [hunk 2]
```diff
@@ -85,6 +85,96 @@ static void _pkg_mgr_event_cb (const char *type, const char *package,
...
+  /**
+   * TODO
+   * 1) Check the directory and xml file of the installed resource package.
+   * 2) If target package has model and xml files but not registered in database, register them.
+   */
+  _I ("Pkg name: %s", pkg_name);
+  pkg_path = g_strdup_printf ("/opt/usr/globalapps/%s/shared/res", pkg_name);
+
+  dir = g_dir_open (pkg_path, 0, NULL);
+  if (dir) {
```
Comment (songgot) [inline daemon/pkg-mgr.c:118]: The NULL case is not considered, did you intend it ?

File: daemon/pkg-mgr.c [hunk 3]
```diff
@@ -85,6 +85,96 @@ static void _pkg_mgr_event_cb (const char *type, const char *package,
...
+    return false;
+  }
+
+  /**
+   * TODO
+   * 1) Check the directory and xml file of the installed resource package.
+   * 2) If target package has model and xml files but not registered in database, register them.
+   */
+  _I ("Pkg name: %s", pkg_name);
+  pkg_path = g_strdup_printf ("/opt/usr/globalapps/%s/shared/res", pkg_name);
```
Comment (anyj0527) [inline daemon/pkg-mgr.c:115]: This would be "/opt/usr/globalapps/%s/res/global/{res-type}"

**Comment by again4you** [comment]
@anyj0527 already applied this patch to his work.

**Pull request closed without merging**